lib/formatters/tapFormatter.mjs

```diff
@@ -1,46 +1,55 @@
+import { isNumber } from '../utils/validateTypes.mjs';
 import preprocessWarnings from './preprocessWarnings.mjs';
 
-/**
- * @type {import('stylelint').Formatter}
- */
+/** @typedef {import('stylelint').Warning} Warning */
+/** @typedef {Array<Omit<Warning, 'rule'>>} Warnings */
+
+/** @type {import('stylelint').Formatter} */
 export default function tapFormatter(results) {
-	const lines = [`TAP version 13\n1..${results.length}`];
+	const lines = ['TAP version 14', `1..${results.length}`];
 
 	for (const [index, result] of results.entries()) {
 		preprocessWarnings(result);
+		// SKIP indicates "that a test was not run, or if it was,
+		// that its success or failure is being temporarily ignored."
+		// skipped tests must not be treated as test failures though
+		// if a file is ignored, errored is currently always false
+		const ignored = result.ignored ? ' # SKIP ignored' : '';
+		const error = result.errored ? 'not ' : '';
 
-		lines.push(
-			`${result.errored ? 'not ok' : 'ok'} ${index + 1} - ${result.ignored ? 'ignored ' : ''}${
-				result.source
-			}`,
-		);
-
-		if (result.warnings.length > 0) {
-			lines.push('---', 'messages:');
-
-			for (const warning of result.warnings) {
-				lines.push(
-					` - message: "${warning.text}"`,
-					`   severity: ${warning.severity}`,
-					`   data:`,
-					`     line: ${warning.line}`,
-					`     column: ${warning.column}`,
-				);
-
-				if (typeof warning.endLine === 'number') {
-					lines.push(`     endLine: ${warning.endLine}`);
-				}
+		lines.push(`${error}ok ${index + 1} - ${result.source}${ignored}`);
 
-				if (typeof warning.endColumn === 'number') {
-					lines.push(`     endColumn: ${warning.endColumn}`);
-				}
+		if (error) {
+			lines.push('  ---');
+
+			/** @type {Record<string, Warnings>} */
+			const rules = {};
```

Comment:
[suggest] I just remembered that Maps have more benefits than Objects, e.g. more straightforward keys order.
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps

So, how about using a Map here?

```suggestion
			/** @type {Map<string, Warnings>} */
			const rules = new Map;
```

Reply:
> more straightforward keys order

That changed circa ES5 IIRC.

> So, how about using a Map here?

I like the conciseness of `const warnings = (rules[name] ??= []);`.
i.e. it would be much more verbose with `get` and `set`

Reply:
That makes sense, too. Okay, let's leave it as-is. 👍🏼